src/nxt_process.c

```diff
@@ -34,84 +40,221 @@ nxt_bool_t  nxt_proc_remove_notify_matrix[NXT_PROCESS_MAX][NXT_PROCESS_MAX] = {
+        nxt_debug(task, "app \"%s\" real pid %d", init->name, nxt_pid);
+        nxt_debug(task, "app \"%s\" isolated pid: %d", init->name, pid);
+    }
 
-        /* Clean inherited cached thread tid. */
-        task->thread->tid = 0;
+    n = read(parentfd, &parent_status, sizeof(parent_status));
+    if (nxt_slow_path(n == -1 || n != sizeof(parent_status))) {
+        nxt_alert(task, "failed to read parent status");
+        return NXT_ERROR;
+    }
 
-        process->pid = nxt_pid;
+    if (nxt_slow_path(close(parentfd) == -1)) {
+        nxt_alert(task, "failed to close reader pipe fd");
+        return NXT_ERROR;
+    }
 
-        ptype = process->init->type;
+    if (nxt_slow_path(parent_status != NXT_OK)) {
+        return parent_status;
+    }
 
-        nxt_port_reset_next_id();
+    ptype = init->type;
 
-        nxt_event_engine_thread_adopt(task->thread->engine);
+    nxt_port_reset_next_id();
 
-        /* Remove not ready processes */
-        nxt_runtime_process_each(rt, p) {
+    nxt_event_engine_thread_adopt(task->thread->engine);
 
-            if (nxt_proc_conn_matrix[ptype][nxt_process_type(p)] == 0) {
-                nxt_debug(task, "remove not required process %PI", p->pid);
+    /* Remove not ready processes */
+    nxt_runtime_process_each(rt, p) {
 
-                nxt_process_close_ports(task, p);
+        if (nxt_proc_conn_matrix[ptype][nxt_process_type(p)] == 0) {
+            nxt_debug(task, "remove not required process %PI", p->pid);
 
-                continue;
-            }
+            nxt_process_close_ports(task, p);
 
-            if (!p->ready) {
-                nxt_debug(task, "remove not ready process %PI", p->pid);
+            continue;
+        }
 
-                nxt_process_close_ports(task, p);
+        if (!p->ready) {
+            nxt_debug(task, "remove not ready process %PI", p->pid);
 
-                continue;
-            }
+            nxt_process_close_ports(task, p);
 
-            nxt_port_mmaps_destroy(&p->incoming, 0);
-            nxt_port_mmaps_destroy(&p->outgoing, 0);
+            continue;
+        }
 
-        } nxt_runtime_process_loop;
+        nxt_port_mmaps_destroy(&p->incoming, 0);
+        nxt_port_mmaps_destroy(&p->outgoing, 0);
 
-        nxt_runtime_process_add(task, process);
+    } nxt_runtime_process_loop;
 
-        nxt_process_start(task, process);
+    nxt_runtime_process_add(task, process);
 
-        process->ready = 1;
+    nxt_process_start(task, process);
 
-        break;
+    process->ready = 1;
 
-    default:
-        /* A parent. */
-        nxt_debug(task, "fork(\"%s\"): %PI", process->init->name, pid);
+    return NXT_OK;
+}
 
-        process->pid = pid;
+nxt_pid_t
+nxt_process_create(nxt_task_t *task, nxt_process_t *process)
+{
+    int                 pipefd[2];
+    nxt_int_t           ret;
+    nxt_pid_t           pid;
+    nxt_process_init_t  *init;
 
-        nxt_runtime_process_add(task, process);
+    init = process->init;
```

Comment:
From buildbot:
> src/nxt_process.c: In function 'nxt_process_create':
src/nxt_process.c:145:26: error: variable 'init' set but not used [-Werror=unused-but-set-variable]
  145 |     nxt_process_init_t  *init;
      |                          ^~~~
cc1: all warnings being treated as errors

Reply:
done